app/services/jobs/notifications/notifications-status-updates.service.js

```diff
@@ -5,58 +5,85 @@
 const UpdateNotificationsService = require('../../notifications/setup/update-notifications.service.js')
-const { timestampForPostgres } = require('../../../lib/general.lib.js')
 
 /**
  * Orchestrates the process of fetching and updating the status of 'scheduledNotification' from the Notify service.
  *
- * This function iterates over a list of 'events' that require their associated 'scheduledNotifications' to have
- * their status checked and updated based on the latest information from Notify.
+ * This function retrieves a list of 'scheduledNotifications' and iterates over them to update their status
+ * according to the latest information from Notify.
+ *
+ * The updates are done in batches to comply with rate-limiting constraints imposed by the Notify service.
+ * The rate limit is set to 3,000 messages per minute, which the service respects during processing.
  *
- * Each 'scheduledNotification' status will be updated individually, adhering to the rate-limiting constraints imposed
- * by the 'BatchNotificationsService' provided by Notify. The rate limit is 3,000 messages per minute.
+ * If the number of 'scheduledNotifications' exceeds the internal batch size limit, the notifications are split into
+ * smaller batches for processing. This ensures efficient processing while adhering to the rate limit.
  *
- * If the number of 'scheduledNotifications' associated with an event exceeds the internal batch size limit, the
- * notifications will be split into smaller batches for processing. This ensures that the service respects the rate
- * limit while processing a large number of notifications.
+ * After all notifications are updated, any notifications with an event that has failed (Notify error, not a 4xx or 5xx
+ * error) will trigger an update to the error count for the corresponding event.
  *
  */
 async function go() {
-  const events = await FetchEventNotificationsService.go()
+  const { batchSize, delay } = NotifyConfig
+
+  const scheduledNotifications = await FetchScheduledNotificationsService.go()
+
+  for (let i = 0; i < scheduledNotifications.length; i += batchSize) {
+    const batchScheduledNotifications = scheduledNotifications.slice(i, i + batchSize)
 
-  for (const event of events) {
-    await _processEvent(event)
+    await _batch(batchScheduledNotifications)
+
+    await _delay(delay)
   }
 }
 
-async function _notificationStatus(scheduledNotification, notifications) {
+async function _batch(scheduledNotifications) {
+  const toUpdateNotifications = _toUpdateNotifications(scheduledNotifications)
+
+  const updatedNotifications = await _updateNotifications(toUpdateNotifications)
+
+  await UpdateNotificationsService.go(updatedNotifications)
+}
+
+async function _delay(delay) {
+  return setTimeout(delay)
+}
+
+async function _notificationStatus(scheduledNotification) {
   const notifyResponse = await NotifyStatusService.go(scheduledNotification.notifyId)
 
-  if (!notifyResponse.errors) {
+  if (notifyResponse.errors) {
```

Comment:
This is necessary to avoid complicated logic to remove any 'undefined' statuses when the promise resolves.